rust/main/hyperlane-base/src/metrics/runtime_metrics.rs

```diff
@@ -0,0 +1,64 @@
+use std::{fmt::Debug, time::Duration};
+
+use eyre::Result;
+use hyperlane_core::metrics::agent::METRICS_SCRAPE_INTERVAL;
+use prometheus::IntGauge;
+use tokio::{task::JoinHandle, time::MissedTickBehavior};
+use tokio_metrics::{TaskMetrics, TaskMonitor};
+use tracing::{info_span, instrument::Instrumented, Instrument};
+
+use super::CoreMetrics;
+
+const RUNTIME_DROPPED_TASKS_HELP: &str = "The number of tasks dropped";
+
+/// Metrics for the runtime
+pub struct RuntimeMetrics {
+    producer: TaskMonitor,
+    dropped_tasks: IntGauge,
+}
+
+// Need this to be included in the agents structs
+impl Debug for RuntimeMetrics {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("RuntimeMetrics").finish()
+    }
+}
+
+impl RuntimeMetrics {
+    pub(crate) fn new(metrics: &CoreMetrics, task_monitor: TaskMonitor) -> Result<RuntimeMetrics> {
+        let dropped_tasks = metrics
+            .new_int_gauge("tokio_dropped_tasks", RUNTIME_DROPPED_TASKS_HELP, &[])?
+            .with_label_values(&[]);
+        let chain_metrics = Self {
+            producer: task_monitor,
+            dropped_tasks,
+        };
+        Ok(chain_metrics)
+    }
+
+    fn update(&mut self, metrics: TaskMetrics) {
+        self.dropped_tasks.add(metrics.dropped_count as i64);
```

Comment:
this `.add`s rather than `.set`s to get the cumulative value over time

Reply:
I wonder why we choose `IntGauge` over `IntCounter`?

Reply:
ah didn't know the gauge is signed while counter is unsigned, will change